### PR TITLE
openssl: fix CVE-2016-2177

### PR DESCRIPTION
**Pull request by rasendubi**
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Note we can't use `fetchpatch` because that leads to cyclic dependency.

### CODE REVIEW

**Comment by fpletz** [comment]
This issue has a very low priority upstream. Even Debian and Red Hat (who don't have the rebuild all dependent packages) didn't update their openssl packages for this issue.

See https://bugzilla.redhat.com/show_bug.cgi?id=1341705 & https://security-tracker.debian.org/tracker/CVE-2016-2177.

So :-1: from me.

File: pkgs/development/libraries/openssl/default.nix [hunk 1]
```diff
@@ -8,7 +8,7 @@ let
   opensslCrossSystem = stdenv.cross.openssl.system or
     (throw "openssl needs its platform name cross building");
 
-  common = { version, sha256 }: stdenv.mkDerivation rec {
+  common = args@{ version, sha256, patches ? [] }: stdenv.mkDerivation rec {
```
Comment (edolstra) [inline pkgs/development/libraries/openssl/default.nix]: Why the `args@`? You can references the patches via `patches` instead of `args.patches`.

**Comment by edolstra** [comment]
We can merge this into `staging` instead.

File: pkgs/development/libraries/openssl/default.nix (hunk 1 above)
Comment (rasendubi) [inline pkgs/development/libraries/openssl/default.nix]: Because it's `rec` and referencing as `patches` will produce infinite recursion.

**Comment by grahamc** [comment]
Has this been merged in to staging?

**Comment by rasendubi** [comment]
@grahamc it hasn't

**Comment by vcunat** [comment]
Now in staging and 16.03.

**Comment by edolstra** [comment]
Why was this backported to 16.03 if it's low impact (see https://github.com/NixOS/nixpkgs/pull/17928#issuecomment-241725982)?

**Comment by vcunat** [comment]
I didn't understand low priority as not worth fixing/cherry-picking.

**Comment by grahamc** [comment]
I think this issue can be closed, since it is good as done?

**Pull request closed without merging**